From d14aa3524df2687f47dd1d8f726182be3eb38f4b Mon Sep 17 00:00:00 2001
From: Hardik T Shah <hardik.t.shah@intel.com>
Date: Tue, 27 Oct 2015 09:22:57 +0900
Subject: [PATCH 1998/2508] ASoC: Intel: Skylake: Add support for module GUIDs

commit baa6f42c4ed1b82d9e3c6c96a28a793b64f24868 from
https://github.com/01org/linux-apollolake-i

The DSP FW specifies loadable modules using GUIDs so add support to
specify the GUIDs from topology

Change-Id: I5e6df0bcea98a4cfbd4d6a4f32a7e11956e50341
Signed-off-by: Hardik T Shah <hardik.t.shah@intel.com>
Signed-off-by: Omair M Abdullah <omair.m.abdullah@intel.com>
Signed-off-by: Jeeja KP <jeeja.kp@intel.com>
Signed-off-by: Subhransu S. Prusty <subhransu.s.prusty@intel.com>
Signed-off-by: Vinod Koul <vinod.koul@intel.com>
Signed-off-by: Mark Brown <broonie@kernel.org>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/9617
Reviewed-by: B, Jayachandran <jayachandran.b@intel.com>
Reviewed-by: Babu, Ramesh <ramesh.babu@intel.com>
Tested-by: Jayanti, Satya Charitardha <satya.charitardha.jayanti@intel.com>
---
 sound/soc/intel/skylake/skl-topology.c       |    4 ++--
 sound/soc/intel/skylake/skl-topology.h       |    2 +-
 sound/soc/intel/skylake/skl-tplg-interface.h |    3 ++-
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/sound/soc/intel/skylake/skl-topology.c b/sound/soc/intel/skylake/skl-topology.c
index f652eb2..66d16ca 100644
--- a/sound/soc/intel/skylake/skl-topology.c
+++ b/sound/soc/intel/skylake/skl-topology.c
@@ -2024,8 +2024,8 @@ static int skl_tplg_widget_load(struct snd_soc_component *cmpnt,
 	mconfig->formats_config.caps_size = dfw_config->caps.caps_size;
 
 	if (dfw_config->is_loadable)
-		memcpy(mconfig->guid, dfw_config->guid,
-					ARRAY_SIZE(dfw_config->guid));
+		memcpy(mconfig->guid, dfw_config->uuid,
+					ARRAY_SIZE(dfw_config->uuid));
 
 	mconfig->m_in_pin = devm_kzalloc(bus->dev, (mconfig->max_in_queue) *
 						sizeof(*mconfig->m_in_pin),
diff --git a/sound/soc/intel/skylake/skl-topology.h b/sound/soc/intel/skylake/skl-topology.h
index ca9fb16..95812a3 100644
--- a/sound/soc/intel/skylake/skl-topology.h
+++ b/sound/soc/intel/skylake/skl-topology.h
@@ -295,7 +295,7 @@ struct skl_gain_data {
 };
 
 struct skl_module_cfg {
-	char guid[SKL_MOD_NAME];
+	char guid[SKL_UUID_STR_SZ];
 	struct skl_module_inst_id id;
 	u8 domain;
 	bool homogenous_inputs;
diff --git a/sound/soc/intel/skylake/skl-tplg-interface.h b/sound/soc/intel/skylake/skl-tplg-interface.h
index dc2e66f..ea1d2cd 100644
--- a/sound/soc/intel/skylake/skl-tplg-interface.h
+++ b/sound/soc/intel/skylake/skl-tplg-interface.h
@@ -202,7 +202,8 @@ struct skl_dfw_pipe {
 } __packed;
 
 struct skl_dfw_module {
-	char guid[SKL_MOD_NAME];
+	char uuid[SKL_UUID_STR_SZ];
+
 	u16 module_id;
 	u16 instance_id;
 	u32 max_mcps;
-- 
1.7.5.4

